src/AlgebraicGeometry/Schemes/CoveredSchemes/Objects/Properties.jl

```diff
@@ -1,29 +1,122 @@
-export is_empty
+export is_empty, is_smooth, is_reduced, is_integral, is_irreducible
 
 ########################################################################
 # Properties of AbsCoveredSchemes                                      #
+#    by referral to the patches                                        #
 ########################################################################
+
+########################################################################
+# (1) Check and store, whether a covered scheme is empty               #
+########################################################################
+@doc Markdown.doc"""
+   is_empty(X::AbsCoveredScheme)
+
+Return the boolean value whether a covered scheme `X` is empty.
+
+"""
 @attr function is_empty(X::AbsCoveredScheme)
   if !isdefined(X, :coverings) 
     return true
   end
   return all(x->isempty(x), all_patches(default_covering(X)))
 end
 
+########################################################################
+# (2) Check and store, whether a covered scheme is smooth              #
+########################################################################
+@doc Markdown.doc"""
+   is_smooth(X::AbsCoveredScheme)
+
+Return the boolean value whether a covered scheme `X` is smooth.
+
+"""
 @attr function is_smooth(X::AbsCoveredScheme)
   if !isdefined(X, :coverings) 
     return true
   end
-  # TODO: This can be optimized! We only need to check the rank of 
+  # TODO: Can this be optimized? We only need to check the rank of
   # the jacobian matrices where we haven't checked in another chart before. 
+  # This is a tradeoff between cost of Jacobian criterion and cost of
+  # optimizing the use of the covering
   return all(x->is_smooth(x), affine_charts(X))
```

Comment:
More idiomatic would be this:
```suggestion
  return all(is_smooth, affine_charts(X))
```

Reply:
Ok, I will keep this in mind for next time, but not change it in this PR (also holds for the analogous suggestions below).